### PR TITLE
Begin the work to update to SuiteSparse 4.0.1

### PR DESCRIPTION
**Pull request by eriknw**
**Not ready to merge yet**

- Updated usage of `GrB_error`.
- Updated import/export
    - still need to to add import/export for bitmaps and "Full"
- Still need to add the `FIRSTI`, `SECONDJ`, `POSITIONJ1`, etc. operators

### CODE REVIEW

**Comment by eriknw** [comment]
@jim22k, Github Actions don't seem to be running as expected.  In particular, `pygraphblas` is always installed.  The `if: contains(${{ matrix.testopts }}, "pygraphblas")` check in "Optional pygraphblas" section doesn't appear to be working.

**Comment by eriknw** [comment]
@jim22k, Github Actions don't seem to be running as expected.  In particular, `pygraphblas` is always installed.  The `if: contains(${{ matrix.testopts }}, "pygraphblas")` check in "Optional pygraphblas" section doesn't appear to be working.

**Comment by jim22k** [comment]
> @jim22k, Github Actions don't seem to be running as expected. In particular, `pygraphblas` is always installed. The `if: contains(${{ matrix.testopts }}, "pygraphblas")` check in "Optional pygraphblas" section doesn't appear to be working.

The reference guide says you don't need `${{ }}` inside the `if`, so maybe that's the issue. Try changing it to:
`if: contains(matrix.testopts, "pygraphblas")`

File: grblas/backends/suitesparse/patch401.diff [hunk 1]
```diff
@@ -0,0 +1,70 @@
+diff --git a/grblas/backends/suitesparse/suitesparse_graphblas_4.0.1.h b/grblas/backends/suitesparse/suitesparse_graphblas_4.0.1.h
```
Comment (jim22k) [inline grblas/backends/suitesparse/patch401.diff:1]: Is there a reason we need the diff file in the repo?

**Review by jim22k — APPROVED** [review]

**Pull request merged**